### PR TITLE
Implemented QueryType View

### PR DESCRIPTION
**Pull request by bdunogier**
> Implements [EZP-24624](https://jira.ez.no/browse/EZP-24624)
> Status: working, reviews accepted
> Note: many of the changed files are for BDD (21 files with 728 additions and 2 deletions). The real code is much smaller :-)

This implements the Query controller according to the [named queries specifications](https://github.com/ezsystems/ezpublish-kernel/blob/master/doc/specifications/proposed/named_queries/developer_documentation.md).

A new Platform controller is added to the Core: `ez_query` (alias for `ezpublish.controller.query`). It exposes three actions: `location`, `content` and `contentInfo`. The views (template, controller, ...) can be configured using the `query_type_view` key.

It also **supports PagerFanta** by default (could be a separate story, it's in a distinct commit). The list variable assigned to the view template is a pager instead of being an array of value objects. The behaviour can be disabled by setting `enable_pager: false` in the view configuration.
### Coverage

The feature is covered by a couple [behat scenarii](https://github.com/ezsystems/ezpublish-kernel/blob/ezp24624-query_controller/eZ/Bundle/EzPublishCoreBundle/Features/View/named_queries.feature):
- _ez_query controller actions can be overridden_
- _When enable_pager is set to false in a QueryType View, then search results are passed to the template as an array_
- _When enable_pager is not specified in a QueryType View, then search results are passed to the template as a pager object_

Given that none of the view feature was covered before, there are probably things that would better covered in a global View context. I'm open to suggestions.
### TODO
- [x] Consider replacing the `xxx_list` variables with `PagerFanta` instances transparently

### CODE REVIEW

**Comment by andrerom** [comment]
+1 when Travis agrees :)

**Comment by bdunogier** [comment]
I might apply these changes to the demobundle so that Travis has something to agree with :-)

**Comment by lolautruche** [comment]
> Consider replacing the `xxx_list` variables

Will only work for pr0n sites? :smile:

**Comment by bdunogier** [comment]
> Will only work for pr0n sites? :smile:

That's actually a good idea ;)

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/Parser/QueryTypeView.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    ->end()
+                ->end()
+            ->end()
+        ->end();
+    }
+
+    public function preMap(array $config, ContextualizerInterface $contextualizer)
+    {
+        parent::preMap($config, $contextualizer); // TODO: Change the autogenerated stub
+    }
```
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/Parser/QueryTypeView.php]: Can be removed I guess

Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/Parser/QueryTypeView.php]: yep

File: eZ/Bundle/EzPublishCoreBundle/Features/View/named_queries.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+         And that configuration matches the QueryType parameter "type" with the value "article"
+         And there is a route "ez_platform_behat_latest_articles"
+         And that route has the default "_controller" set to "ez_query:content"
+         And that route has the default "queryTypeName" set to "EzPlatformBehatBundle:LatestContent"
+         And that route has the default "viewType" set to "full"
+         And that route has the default "parameters" set to an array with the key "type" set to "article"
+        When I go to that route
+        Then I see the QueryType results listed with this template
+
+
```
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/Features/View/named_queries.feature]: Extra empty line

File: eZ/Publish/Core/MVC/Symfony/View/ParametersInjector/QueryTypePager.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * Configures the PagerFanta instance of a QueryTypeView, if enabled.
+ */
+class QueryTypePager implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [Events::VIEW_PARAMETERS_INJECTION => 'configurePager'];
+    }
+
+    public function configurePager(ViewParametersFilterEvent $event)
```
Comment (emodric) [inline eZ/Publish/Core/MVC/Symfony/View/ParametersInjector/QueryTypePager.php]: Missing implementation, after both returns? :)

Comment (bdunogier) [inline eZ/Publish/Core/MVC/Symfony/View/ParametersInjector/QueryTypePager.php]: Actually it's not used. Unevened spike :)

**Pull request closed without merging**